src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -297,10 +298,103 @@ void snc_quota_manager::maybe_arm_balancer_timer() {
     _balancer_timer.arm(arm_until);
 }
 
+using shard_count_t
+  = boost::uint_t<std::numeric_limits<ss::shard_id>::digits>::fast;
+
+struct borrower_t {
+    ss::shard_id shard_id;
+    // can be 0 or 1, made shard_count_t for easier aggregation
+    inoutpair<shard_count_t> is_borrower;
```

Comment:
The comment and data type aren't clear. `is_borrower` makes me think this can be 0/1 (aligned with the comment) but the data type is `shard_count_t` making me think its a number of shards which are borrowers?

Reply:
yeh, seems like this could be inoutpair<bool>?

Reply:
it's generally better to have types that make sense. why wouldn't you use convert is_borrower to a number in the lambda here? otoh, if it is a count, then name it that.